ui/pages/Swap.tsx

```diff
@@ -1,24 +1,148 @@
 import React, { ReactElement, useCallback, useState } from "react"
+import { BigNumber, utils as ethersUtils } from "ethers"
+import { fetchJson } from "@ethersproject/web"
+import logger from "@tallyho/tally-background/lib/logger"
+import { Asset } from "@tallyho/tally-background/assets"
+import { selectAccountAndTimestampedActivities } from "@tallyho/tally-background/redux-slices/accounts"
 import CorePage from "../components/Core/CorePage"
 import SharedAssetInput from "../components/Shared/SharedAssetInput"
 import SharedButton from "../components/Shared/SharedButton"
 import SharedSlideUpMenu from "../components/Shared/SharedSlideUpMenu"
 import SwapQoute from "../components/Swap/SwapQuote"
 import SharedActivityHeader from "../components/Shared/SharedActivityHeader"
 import SwapTransactionSettings from "../components/Swap/SwapTransactionSettings"
+import { useBackgroundSelector } from "../hooks"
+
+interface SwapAmount {
+  from: string
+  to: string
+}
+
+interface TradingPair {
+  from?: Asset
+  to?: Asset
+  price: BigNumber
+}
+
+interface ZrxToken {
+  symbol: string
+  price: string
+}
 
 export default function Swap(): ReactElement {
   const [openTokenMenu, setOpenTokenMenu] = useState(false)
-  const [selectedCount, setSelectedCount] = useState(0)
+  const [swapTokens, setSwapTokens] = useState<Asset[]>([])
+  const [swapAmount, setSwapAmount] = useState<SwapAmount>({
+    from: "0",
+    to: "0",
+  })
+
+  const [swap, setSwap] = useState<TradingPair>({
+    from: undefined,
+    to: undefined,
+    price: BigNumber.from("0"),
+  })
+
+  const { combinedData } = useBackgroundSelector(
+    selectAccountAndTimestampedActivities
+  )
+
+  const displayAssets = combinedData.assets.map(({ asset }) => asset)
```

Comment:
The selector function should probably yield the display assets directly so it doesn't pollute scope:

```suggestion
  const displayAssets = useBackgroundSelector((state) => {
    const {
      combinedData: { assets },
    } = selectAccountAndTimestampedActivities(state)

    return assets.map(({ asset }) => asset)
  })
```

This also points to the obvious fact that `assets` should be named `assetBalances` in `CombinedAccountData` lol.